src/components/Auth/RequireAuth.js

```diff
@@ -17,21 +17,36 @@ const requireAuth = (WrappedComponent) => {
         }
 
         render() {
-            if(this.props.authState === AuthStates.STUDENT) {
+            var authorized = true;
+
+            // Checks if user has the required permissions
```

Comment:
Components that use RequireAuth pass in a Javascript object (`requiredState`) that contains the permissions needed to access them. This loop is checking the user's permissions from the Redux store against `requiredState`.